File: 2019/1/main.py
from math import floor

with open("input.txt") as f:
    _input = [float(x) for x in f.readlines()]

def calc_fuel(x:float):
    return floor((x / 3)) - 2

answer_1 = sum(
    calc_fuel(x) for x in _input
)
print(answer_1)

answer_2 = 0
for fuel in _input:
    while True:
        fuel = calc_fuel(fuel)
        if fuel < 1: break
        answer_2 += fuel

print(answer_2)

File: 2019/1/main.hs

calcFuel :: (RealFrac a, Integral b) => a -> b
calcFuel input = (floor $ input / 3) - 2

calcAllFuel :: (RealFrac a, Integral b) => b -> a -> b
calcAllFuel acc input = if x < 1
                        then acc
                        else calcAllFuel (acc + x) (realToFrac x)
  where
    x = calcFuel input

main = do
  lines <- fmap lines (readFile "input.txt")
  let masses = map read lines :: [Float]
  print $ sum $ map calcFuel masses
  print $ sum $ map (calcAllFuel 0) masses
